src/test/scala/cognite/spark/v1/RawTableRelationTest.scala

```diff
@@ -682,7 +682,31 @@ class RawTableRelationTest
 
     assert(df.count() == 1)
 
-    val rowsRead = getNumberOfRowsRead(metricsPrefix, f"raw.spark-test-database.${tableName}.rows")
+    val rowsRead = getNumberOfRowsRead(metricsPrefix, s"raw.spark-test-database.$tableName.rows")
+    assert(rowsRead == 1)
+  }
+
+  it should "support pushdown key filters with AND and IN resulting in an empty set of keys" taggedAs ReadTest in {
+    val tableName = "with-boolean-empty-str"
+    val metricsPrefix = s"pushdown.raw.key.${shortRandomString()}"
+    val df = rawRead(tableName, metricsPrefix = Some(metricsPrefix))
+      .where("key in ('k1') and key in ('k2')")
+
+    assert(df.count() == 0)
+
+    // No rows should have been read, so the metric should not exist.
+    a[NullPointerException] should be thrownBy getNumberOfRowsRead(metricsPrefix, s"raw.spark-test-database.$tableName.rows")
+  }
+
+  it should "support pushdown key filters with OR and IN" taggedAs ReadTest in {
+    val tableName = "with-boolean-empty-str"
+    val metricsPrefix = s"pushdown.raw.key.${shortRandomString()}"
+    val df = rawRead(tableName, metricsPrefix = Some(metricsPrefix))
+      .where("bool = 'foo' OR key in ('k2')")
+
+    assert(df.count() == 1)
+
+    val rowsRead = getNumberOfRowsRead(metricsPrefix, s"raw.spark-test-database.$tableName.rows")
     assert(rowsRead == 1)
```

Comment:
Same, if `bool` is in the OR, `key` shouldn't be pushed down
```suggestion
      .where("key = 'random-invalid-key' OR key in ('k2')")

    assert(df.count() == 1)

    val rowsRead = getNumberOfRowsRead(metricsPrefix, s"raw.spark-test-database.$tableName.rows")
    assert(rowsRead == 1)
```